release notes/v0.31.0.md

```diff
@@ -0,0 +1,82 @@
+k6 v0.31.0 is here! :tada: It's a smaller release with some significant performance improvements, a new `http_req_failed` metric and changes to the output subsystem that enable output extensions with [xk6](https://github.com/k6io/xk6)!
+
+## New features
+
+### Output cleanup and registration API ([#1874](https://github.com/loadimpact/k6/pull/1874))
+
+The state of k6's output packages has been a development pain point for a long time which made it difficult to add new outputs in a consistent way. In the refactor done in v0.31.0 this has been mostly addressed and outputs now implement a saner `Output` interface.
+
+In addition to this, an output registry was added to allow external outputs to be easily added. This expands the possibilities of what [xk6](https://github.com/k6io/xk6) can do, so now besides allowing external JS modules, it's also possible to build custom k6 binaries that use external output implementations! This is very useful if you use a system that's currently not supported by [the built-in outputs](https://k6.io/docs/getting-started/results-output), or need some custom handling of the metrics k6 produces.
+
+We are working on the proper documentation for this, as well as the overdue xk6 documentation about JS extensions, so keep a lookout on https://k6.io/docs/ .
+
+
+### Marking requests as failed ([#1856](https://github.com/loadimpact/k6/pull/1856))
+
+It's now possible to declare expected HTTP response statuses for either the entire test or for individual HTTP requests, and k6 will emit a new `http_req_failed` metric as well as tag HTTP metrics with `expected_response: <bool>`.
+
+For example:
+
+```javascript
+import http from 'k6/http';
+
+// Set expected statuses globally for all requests.
+http.setResponseCallback(http.expectedStatuses({min: 200, max: 399}, 418));
+
+export default function () {
+  // This request will be marked as failed.
+  http.get('https://httpbin.test.k6.io/status/400');
+  // This request will be considered as "passed" because of the responseCallback override.
+  http.get('https://httpbin.test.k6.io/status/400', { responseCallback: http.expectedStatuses(400) });
+}
+```
+
+Running this script will produce a summary like:
+
+```
+http_req_duration..............: avg=204.57ms min=203.31ms med=204.57ms max=205.82ms p(90)=205.57ms p(95)=205.7ms
+  { expected_response:true }...: avg=203.31ms min=203.31ms med=203.31ms max=203.31ms p(90)=203.31ms p(95)=203.31ms
+http_req_failed................: 50.00% ✓ 1   ✗ 1
+```
+
+Note the new `http_req_duration` sub-metric for passed requests only and the `http_req_failed` Rate metric.
```

Comment:
```suggestion
Note the new `http_req_duration` sub-metric for expected responses only, and the new `http_req_failed` `Rate` metric. This new new metric and metric tag have many potential use cases, though one of the most important ones is the ability to set better [thresholds](https://k6.io/docs/using-k6/thresholds). For example:
- `'http_req_failed': ['rate<0.1']`, i.e. fail the test if more than 10% of requests fail.
- `'http_req_duration{expected_response:true}': ['p(95)<300', 'p(99.9)<500']` - fail the test if the the 95th percentile HTTP request duration is above 300ms or the 99.9th percentile is above 500ms; specifying `expected_response:true` here may be important, because a lot of times failed requests may return more quickly than normal ones, thus skewing the results and wrongly satisfying the threshold.
```